packages/provider/provider.ts

```
@@ -0,0 +1,122 @@
+
+export const LIFECYCLE = {
+  Singleton: "singleton",
+  Scoped: "scoped",
+  Transient: "transient"
+} as const;
+
+export type Lifecycle = typeof LIFECYCLE[keyof typeof LIFECYCLE];
+
+export class Provider {
+  private static singletons = new Map<InjectionToken, any>();
+  private scoped = new Map<InjectionToken, any>();
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)